hazelcast-jet-core/src/main/java/com/hazelcast/jet/Jet.java

```diff
@@ -258,11 +259,18 @@ private static void assertHazelcastVersion() {
             Properties p = new Properties();
             p.load(resource);
             String jetHzVersion = p.getProperty("jet.hazelcast.version");
+
             if (!hzVersion.equals(jetHzVersion)) {
-                throw new JetException("Jet uses Hazelcast IMDG version " + jetHzVersion + " however " +
+                String message = "Jet uses Hazelcast IMDG version " + jetHzVersion + " however " +
                         "version " + hzVersion + " was found in the classpath. " +
                         " As Jet already shades Hazelcast jars there is no need to explicitly " +
-                        "add a dependency to it.");
+                        "add a dependency to it.";
+                boolean errorOnMismatch = parseBoolean(System.getProperty("jet.hazelcast.errorOnMismatch", "true"));
```

Comment:
We have `com.hazelcast.jet.core.JetProperties` that declares and documents all the user-configurable properties Jet recognizes.

The user can configure a property both as a system property and in the configuration file, this would be an exception. We could either make the check in a place where the properties are accessible, or failing that, document that this only works as a system property.

Also, please adjust the name to comply with the convention, and semantics so that the default value of `false` does the right thing: `jet.imdg.version.mismatch.check.disabled`.

Reply:
Thanks, I didn't know about it. Will submit changes shortly

Reply:
We discussed this some more, we already have a precedent for a property that only works as a sysprop: `hazelcast.logging.type`, and the reasons are similar -- it's read at class init time. So I think we're good if we just make a note on this in the docs.